product_pricelist_supplierinfo/models/product_template.py

```diff
@@ -13,41 +13,16 @@ def _get_supplierinfo_pricelist_price(
             self, rule, date=None, quantity=None, product_id=None):
         """Method for getting the price from supplier info."""
         self.ensure_one()
+        product = self.product_variant_id
         if product_id:
-            domain = [
-                '|',
-                ('product_id', '=', product_id),
-                ('product_tmpl_id', '=', self.id),
-            ]
-        else:
-            domain = [
-                ('product_tmpl_id', '=', self.id),
-            ]
-        if not rule.no_supplierinfo_min_quantity and quantity:
-            domain += [
-                '|',
-                ('min_qty', '=', False),
-                ('min_qty', '<=', quantity),
-            ]
-        if date:
-            domain += [
-                '|',
-                ('date_start', '=', False),
-                ('date_start', '<=', date),
-                '|',
-                ('date_end', '=', False),
-                ('date_end', '>=', date),
-            ]
-        # We use a different default order because we are interested in getting
-        # the price for lowest minimum quantity if no_supplierinfo_min_quantity
-        supplierinfos = self.env['product.supplierinfo'].search(
-            domain, order='min_qty,sequence,price',
-        )
+            product = product.browse(product_id)
         if rule.no_supplierinfo_min_quantity:
-            selected_supplierinfo = supplierinfos[:1]
+            quantity = 1.0
+        seller = product._select_seller(quantity=quantity, date=date)
+        if seller:
+            price = seller._get_supplierinfo_pricelist_price()
         else:
-            selected_supplierinfo = supplierinfos[-1:]
-        price = selected_supplierinfo._get_supplierinfo_pricelist_price()
+            price = 0.0
```

Comment:
Hi @ruuter I think you can delete this `else` and set the price before the seller, something like this:

```python
price = 0.0
seller = product._select_seller(quantity=quantity, date=date)
if seller:
    price = seller._get_supplierinfo_pricelist_price()
```
This way you avoid the need of doing another test to cover it and the warning of the coverage will be discarded

Reply:
Hmm okay thanks! Did now know it affects coverage.

Reply:
You're welcome :smiley: It was affecting because there wasn't a test to cover the else so the line was not covered and a warning appeared but now it is covered and no warning is shown